### PR TITLE
Add support for Lint endpoints

### PR DESCRIPTION
**Pull request by mchanyeechoy**
Add endpoints from https://docs.gitlab.com/ee/api/lint.html

### CODE REVIEW

File: NGitLab/Impl/LintClient.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    internal class LintClient : ILintClient
+    {
+        private readonly API _api;
+
+        public LintClient(API api)
+        {
+            _api = api;
+        }
+
+        public Task<LintCI> LintGitLabCIYamlAsync(string projectId, string yamlContent, LintCIOptions options, CancellationToken cancellationToken = default)
```
Comment (louis-z) [inline NGitLab/Impl/LintClient.cs]: I think the methods can have clearer names.

In [the doc](https://docs.gitlab.com/ee/api/lint.html#validate-a-ci-yaml-configuration-with-a-namespace), this is described as `Validate a CI YAML configuration with a namespace`, but to make it even clearer that we're passing in YAML content:

```suggestion
        public Task<LintCI> ValidateCIYamlContentAsync(string projectId, string yamlContent, LintCIOptions options, CancellationToken cancellationToken = default)
```

File: NGitLab/Impl/LintClient.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            var url = BuildLintCIUrl(projectId, options);
+            url = Utils.AddParameter(url, "content", yamlContent);
+
+            return _api.Post().ToAsync<LintCI>(url, cancellationToken);
+        }
+
+        public Task<LintCI> LintProjectGitLabCIYamlAsync(string projectId, LintCIOptions options, CancellationToken cancellationToken = default)
```
Comment (louis-z) [inline NGitLab/Impl/LintClient.cs]: The [doc](https://docs.gitlab.com/ee/api/lint.html#validate-a-projects-ci-configuration) calls this `Validate a project’s CI configuration`

```suggestion
        public Task<LintCI> ValidateProjectCIConfigurationlAsync(string projectId, LintCIOptions options, CancellationToken cancellationToken = default)
```

File: NGitLab/Impl/LintClient.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+        public LintClient(API api)
+        {
+            _api = api;
+        }
+
+        public Task<LintCI> LintGitLabCIYamlAsync(string projectId, string yamlContent, LintCIOptions options, CancellationToken cancellationToken = default)
+        {
+            var url = BuildLintCIUrl(projectId, options);
+            url = Utils.AddParameter(url, "content", yamlContent);
```
Comment (louis-z) [inline NGitLab/Impl/LintClient.cs]: According to the request example in the doc, `yamlContent` is supposed to be passed in as POST data, not as query parameter:

> curl --header "Content-Type: application/json" "https://gitlab.example.com/api/v4/projects/:id/ci/lint" --data '{"content": "{ \"image\": \"ruby:2.6\", \"services\": [\"postgres\"], \"before_script\": [\"bundle install\", \"bundle exec rake db:create\"], \"variables\": {\"DB_NAME\": \"postgres\"}, \"types\": [\"test\", \"deploy\", \"notify\"], \"rspec\": { \"script\": \"rake spec\", \"tags\": [\"ruby\", \"postgres\"], \"only\": [\"branches\"]}}"}'

It's strange that tests are passing... 🤔

File: NGitLab/Models/LintCI.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+namespace NGitLab.Models
+{
+    public class LintCI
+    {
+        public const string Url = "/ci/lint";
+
+        [JsonPropertyName("valid")]
+        public bool Valid { get; set; }
+
+        [JsonPropertyName("id")]
```
Comment (louis-z) [inline NGitLab/Models/LintCI.cs]:
```suggestion
        [JsonPropertyName("merged_yaml")]
```

File: NGitLab/Models/LintCIOptions.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+﻿namespace NGitLab.Models
+{
+    public class LintCIOptions
+    {
+        /// <summary>
+        /// Run pipeline creation simulation, or only do static check.
+        /// </summary>
+        public bool DryRun { get; set; }
```
Comment (louis-z) [inline NGitLab/Models/LintCIOptions.cs]: Since it's not a required param, can you make it nullable?

**Review by louis-z — CHANGES_REQUESTED** [review]

File: NGitLab/Impl/LintClient.cs (hunk 3 above)
Comment (mchanyeechoy) [reply NGitLab/Impl/LintClient.cs]: Seems like they both work, but I will change it to be inline with the docs 🙂

**Review by louis-z — APPROVED** [review]

**Pull request merged**